### PR TITLE
Added in a flag to control whether or not the InboundClaimTypeMap is used.

### PR DESCRIPTION
**Pull request by mafurman**
Addresses #923.

Also simplified/refactored the Actor test in the JwtSecurityHandlerTests.

### CODE REVIEW

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 1]
```diff
@@ -1112,6 +1147,42 @@ protected virtual ClaimsIdentity CreateClaimsIdentity(JwtSecurityToken jwtToken,
...
+        private ClaimsIdentity CreateClaimsIdentityWithoutMapping(JwtSecurityToken jwtToken, string actualIssuer, TokenValidationParameters validationParameters)
+        {
+            ClaimsIdentity identity = validationParameters.CreateClaimsIdentity(jwtToken, actualIssuer);
+            foreach (Claim jwtClaim in jwtToken.Claims)
+            {
+                if (_inboundClaimFilter.Contains(jwtClaim.Type))
+                    continue;
+
+                string claimType = jwtClaim.Type;
+
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: remove this line

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs [hunk 2]
```diff
@@ -1112,6 +1147,42 @@ protected virtual ClaimsIdentity CreateClaimsIdentity(JwtSecurityToken jwtToken,
...
+                    if (CanReadToken(jwtClaim.Value))
+                    {
+                        JwtSecurityToken actor = ReadToken(jwtClaim.Value) as JwtSecurityToken;
+                        identity.Actor = CreateClaimsIdentity(actor, actualIssuer, validationParameters);
+                    }
+                }
+
+                Claim claim = new Claim(claimType, jwtClaim.Value, jwtClaim.ValueType, actualIssuer, actualIssuer, identity);
+
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: can remove this line.

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs (hunk 1 above)
Comment (mafurman) [reply src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: 👍

File: src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs (hunk 2 above)
Comment (mafurman) [reply src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs]: 👍

**Pull request merged**